collect_app/src/androidTest/java/org/odk/collect/android/support/pages/ProjectSettingsDialogPage.kt

```diff
@@ -42,8 +42,13 @@ internal class ProjectSettingsDialogPage(rule: ActivityTestRule<*>) : Page<Proje
         return AddProjectDialogPage(rule).assertOnPage()
     }
 
-    fun assertProjectIconColor(color: String): ProjectSettingsDialogPage {
-        Espresso.onView(withId(R.id.project_icon)).check(matches(withBackgroundColor(Color.parseColor(color))))
+    fun assertCurrentProject(project: Project): ProjectSettingsDialogPage {
+        Espresso.onView(withId(R.id.current_project)).check(matches(withProject(project)))
```

Comment:
Again it would be nice to avoid `findViewById` in these methods, but it seems like a hard one. Sometimes that can be a hint that we've created an accessibility problem in our app - if we don't have a way to match uniquely on text (or content description) then it also means the screen would be ambiguous when read through a screen reader (it's interesting to think of our UI tests as using a screen reader as they need to do everything through text).

I just had a play around with the account settings dialog in Google's Contacts app with TalkBack enabled and when I clicked on the current account it said "Signed in as <my main account>" and when I tapped on one of the other accounts it said "Use <other account>". We could do something similar and have a `contentDescription` on the project item - for the current account it could be "Using <project name> project" and for others it could be "Switch to <project name> project". This way we'd have something to match on in Espresso (using `withContentDescription`) and also be supporting screen readers.

Reply:
You mean replacing `withId` with `withContentDescription`? Should we start using `contentDescription` more often? I thought about it and I guessed it's not important in our app so I'm not sure.

Reply:
Yeah there are other projects where I've just used `withContentDecription` as the main text matcher (instead of `withText`). It **usually** works exactly the same as you automatically get content descriptions for `TextView`s. For me, it's a win-win as it helps our app be more inclusive to people that use screen readers but also makes it easier to write tests for cases where we have icon buttons or list items that perform an action on click (neither of which would always be labelled with text).

Reply:
The drawback is that I would need to set it grammatically because if I added `android:contentDescription` to those layouts it's not visible for Espresso. Probably it works for single elements like textViews, imageViews in layouts only?